pkg/storage/scrub.go

```diff
@@ -0,0 +1,244 @@
+package storage
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/anuvu/zot/errors"
+	"github.com/olekukonko/tablewriter"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/opencontainers/umoci"
+	"github.com/opencontainers/umoci/oci/casext"
+)
+
+const (
+	imageNameWidth = 32
+	tagWidth       = 24
+	statusWidth    = 8
+	errorWidth     = 8
+
+	colImageNameIndex = 0
+	colTagIndex       = 1
+	colStatusIndex    = 2
+	colErrorIndex     = 3
+)
+
+type ScrubImageResult struct {
+	ImageName string `json:"image_name"`
+	Tag       string `json:"tag"`
+	Status    string `json:"status"`
+	Error     string `json:"error"`
+}
+
+type ScrubResults struct {
+	ScrubResults []ScrubImageResult `json:"scrub_results"`
+}
+
+func (sc StoreController) CheckAllBlobsIntegrity() (ScrubResults, error) {
+	imagesList := []ImageStore{}
+	imagesList = append(imagesList, sc.DefaultStore)
+
+	for _, subStore := range sc.SubStore {
+		imagesList = append(imagesList, subStore)
+	}
+
+	results := ScrubResults{}
+
+	for _, is := range imagesList {
```

Comment:
Lines 47-53 can be combined into a single loop?

Reply:
In order to combine this in a single loop this is what I did: First of all, I need to iterate through all repos from controller, this means the DefaultStore which has type ImageStore and also all values from SubStore which is a map[string]ImageStore. So to iterate through all of this in a single loop I created a map[string]ImageStore using SubStore's elements if SubStore is not nil, and after this I added to it a new entry with key  "" and value DefaultSore.